cloud/scope/powervs_cluster.go

```diff
@@ -1709,57 +1728,131 @@ func (s *PowerVSClusterScope) checkTransitGatewayStatus(tg *tgapiv1.TransitGatew
-	requeue := false
+// checkAndUpdateTransitGatewayConnections checks given transit gateway's connections status.
+// // if update is set to true, it updates the transit gateway connections too if it is not exist already.
+func (s *PowerVSClusterScope) checkAndUpdateTransitGatewayConnections(transitGateway *tgapiv1.TransitGateway, update bool) (bool, *infrav1beta2.ResourceReference, *infrav1beta2.ResourceReference, error) {
 	tgConnections, _, err := s.TransitGatewayClient.ListTransitGatewayConnections(&tgapiv1.ListTransitGatewayConnectionsOptions{
-		TransitGatewayID: id,
+		TransitGatewayID: transitGateway.ID,
 	})
 	if err != nil {
-		return requeue, fmt.Errorf("failed to list transit gateway connections: %w", err)
-	}
-
-	if len(tgConnections.Connections) == 0 {
-		return requeue, fmt.Errorf("no connections are attached to transit gateway")
+		return false, nil, nil, fmt.Errorf("failed to list transit gateway connections: %w", err)
 	}
 
 	vpcCRN, err := s.fetchVPCCRN()
 	if err != nil {
-		return requeue, fmt.Errorf("failed to fetch VPC CRN: %w", err)
+		return false, nil, nil, fmt.Errorf("failed to fetch VPC CRN: %w", err)
 	}
 
 	pvsServiceInstanceCRN, err := s.fetchPowerVSServiceInstanceCRN()
 	if err != nil {
-		return requeue, fmt.Errorf("failed to fetch PowerVS service instance CRN: %w", err)
+		return false, nil, nil, fmt.Errorf("failed to fetch PowerVS service instance CRN: %w", err)
 	}
 
+	var powerVSConnResource, vpcConnResource *infrav1beta2.ResourceReference
+
+	if len(tgConnections.Connections) == 0 {
+		if update {
+			s.V(3).Info("Connections not exist on existing transit gateway, creating them")
+			powerVSConnID, vpcConnID, err := s.createTransitGatewayConnections(transitGateway, pvsServiceInstanceCRN, vpcCRN)
+			if err != nil {
+				return false, nil, nil, err
+			}
+
+			powerVSConnResource = &infrav1beta2.ResourceReference{
+				ID:                powerVSConnID,
+				ControllerCreated: ptr.To(true),
+			}
+
+			vpcConnResource = &infrav1beta2.ResourceReference{
+				ID:                vpcConnID,
+				ControllerCreated: ptr.To(true),
+			}
+
+			return true, powerVSConnResource, vpcConnResource, nil
+		}
+
+		return false, nil, nil, fmt.Errorf("no connections are attached to transit gateway")
+	}
+
+	var requeue, powerVSAttached, vpcAttached bool
+	requeue, powerVSAttached, vpcAttached, err = s.validateTransitGatewayConnections(tgConnections.Connections, vpcCRN, pvsServiceInstanceCRN)
+	if err != nil {
+		return requeue, nil, nil, err
+	} else if requeue {
+		return requeue, nil, nil, nil
+	}
+
+	if !update && (!powerVSAttached || !vpcAttached) {
+		return false, nil, nil, fmt.Errorf("either one of PowerVS or VPC transit gateway connections is not attached, PowerVS: %t VPC: %t", powerVSAttached, vpcAttached)
+	}
+
+	if powerVSAttached && vpcAttached {
+		if s.IBMPowerVSCluster.Status.TransitGateway == nil {
+			vpcConnResource = &infrav1beta2.ResourceReference{
+				ControllerCreated: ptr.To(false),
```

Comment:
We intentionally not storing connections id here is status right? also if we dont set anything the default value for controller created is also false, but I think explicitly setting will enhance readability of code.